app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala

```diff
@@ -112,10 +111,13 @@ class SecureMessageController @Inject()(
           case _ =>
             authorised()
               .retrieve(Retrievals.allEnrolments) { authEnrolments =>
-                val filteredEnrolments = authEnrolments.filter(enrolmentKeys, customerEnrolments) //TODO: move this to service
-                secureMessageService.getConversationsFiltered(filteredEnrolments, tags).flatMap { conversationDetails =>
-                  Future.successful(Ok(Json.toJson(conversationDetails)))
-                }
+                secureMessageService
+                  .getConversationsFiltered(
+                    authEnrolments,
+                    ConversationFilters(enrolmentKeys, customerEnrolments, tags))
+                  .flatMap { conversationDetails =>
```

Comment:
Can you not map and return Ok(Json.toJson(conversationDetails))

Reply:
@jim-collins it was there before, but good point, changing it now.